apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex

```diff
@@ -190,6 +193,37 @@ defmodule Lexical.RemoteControl.CodeIntelligence.Entity do
     end
   end
 
+  defp maybe_pad_phoenix_scope_module(module_string, charlist, analysis, position) do
+    with true <- controller_module_at_cursor?(charlist),
+         scope_segments = phoenix_scope_segments(analysis, position),
+         false <- is_nil(scope_segments),
+         {:ok, module} <-
+           RemoteControl.Analyzer.expand_alias(scope_segments, analysis, position) do
+      [module, module_string] |> Module.concat() |> inspect()
+    else
+      _ ->
+        module_string
+    end
+  end
+
+  defp phoenix_scope_segments(analysis, position) do
+    path = Ast.path_at(analysis, position)
+    scope_path = Future.Macro.path(path, &match?({:scope, _, [_ | _]}, &1))
+
+    if scope_path do
+      {:scope, _, [_, {:__aliases__, _, segments} | _]} = hd(scope_path)
+      segments
+    end
+  end
+
+  defp controller_module_at_cursor?(charlist) do
+    charlist |> last_module_at_position() |> String.ends_with?("Controller")
```

Comment:
this heuristic isn't particularly good, lots of things can end with `Controller` and not be phoenix controllers. We have access to these modules and should check for certain functions inside them before deeming them to be controllers.

Reply:
So, rather than look for a trailing `Controller` find some of these functions

https://hexdocs.pm/phoenix/Phoenix.Controller.html#functions

Reply:
> So, rather than look for a trailing Controller find some of these functions

But we can't do this before we expand the `alias` of the scope. Since we have `fetch_phoenix_scope_xx`for further checks, then is doing an `ends_with` check enough? 

Of course, we can rename this function to make it clearer.

Reply:
You can do that, get the expected module. Then see if it has one of those functions exported

Reply:
> You can do that, get the expected module.

What I mean is, before I know the `alias` of the scope, I can't obtain the specific module. The current module is a `Router` module, not a certain `Controller`.

Reply:
Don't you have the pieces in the parent scope and the current route?

Reply:
Okay, I think I understand what you mean now. You're suggesting to do this check **after** the fetch. I'll make some modifications in the next few days.

Additionally, I think we also need to consider the issues with *nested scopes* and the `LiveView` modules.

Reply:
Additionally, regarding this kind of function_exported check, do you have any suggestions for testing? It seems I can't simply use a patch to simulate it.

Reply:
Can you not patch `Kernel.function_exported?/2`